exporter/trace/jaeger/jaeger.go

```diff
@@ -0,0 +1,340 @@
+	// For example, localhost:6831.
+	AgentEndpoint string
+
+	// OnError is the hook to be called when there is
+	// an error occurred when uploading the stats data.
+	// If no custom hook is set, errors are logged.
+	// Optional.
+	OnError func(err error)
+
+	// Username to be used if basic auth is required.
+	// Optional.
+	Username string
+
+	// Password to be used if basic auth is required.
+	// Optional.
+	Password string
+
+	// Process contains the information about the exporting process.
+	Process Process
+
+	//BufferMaxCount defines the total number of traces that can be buffered in memory
+	BufferMaxCount int
+}
+
+// NewExporter returns a trace.Exporter implementation that exports
+// the collected spans to Jaeger.
+func NewExporter(o Options) (*Exporter, error) {
+	if o.CollectorEndpoint == "" && o.AgentEndpoint == "" {
+		return nil, errors.New("missing endpoint for Jaeger exporter")
+	}
+
+	var endpoint string
+	var client *agentClientUDP
+	var err error
+	if o.CollectorEndpoint != "" {
+		endpoint = o.CollectorEndpoint
+	} else {
+		client, err = newAgentClientUDP(o.AgentEndpoint, udpPacketMaxLength)
+		if err != nil {
+			return nil, err
+		}
+	}
+	onError := func(err error) {
+		if o.OnError != nil {
+			o.OnError(err)
+			return
+		}
+		log.Printf("Error when uploading spans to Jaeger: %v", err)
+	}
+	service := o.Process.ServiceName
+	if service == "" {
+		service = defaultServiceName
+	}
+	tags := make([]*gen.Tag, len(o.Process.Tags))
+	for i, tag := range o.Process.Tags {
+		tags[i] = attributeToTag(tag.key, tag.value)
+	}
+	e := &Exporter{
+		endpoint:      endpoint,
+		agentEndpoint: o.AgentEndpoint,
+		client:        client,
+		username:      o.Username,
+		password:      o.Password,
+		process: &gen.Process{
+			ServiceName: service,
+			Tags:        tags,
+		},
+	}
+	bundler := bundler.NewBundler((*gen.Span)(nil), func(bundle interface{}) {
+		if err := e.upload(bundle.([]*gen.Span)); err != nil {
+			onError(err)
+		}
+	})
+
+	// Set BufferedByteLimit with the total number of spans that are permissible to be held in memory.
+	// This needs to be done since the size of messages is always set to 1. Failing to set this would allow
+	// 1G messages to be held in memory since that is the default value of BufferedByteLimit.
+	if o.BufferMaxCount != 0 {
+		bundler.BufferedByteLimit = o.BufferMaxCount
+	}
+
+	e.bundler = bundler
+	return e, nil
+}
+
+// Process contains the information exported to jaeger about the source
+// of the trace data.
+type Process struct {
+	// ServiceName is the Jaeger service name.
+	ServiceName string
+
+	// Tags are added to Jaeger Process exports
+	Tags []Tag
+}
+
+// Tag defines a key-value pair
+// It is limited to the possible conversions to *jaeger.Tag by attributeToTag
+type Tag struct {
+	key   string
+	value interface{}
+}
+
+// BoolTag creates a new tag of type bool, exported as jaeger.TagType_BOOL
+func BoolTag(key string, value bool) Tag {
```

Comment:
Looks like this and the other Tag constructors here are not used? I see `attributeToTag` might have once used them? Can we remove this `Tag`?

Reply:
removed.